File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = {}
        result = []
        for i in strs:
            key = tuple(sorted(i))
            if key not in anagrams:
                anagrams[key] = []
            anagrams[key].append(i)
        return anagrams.values()

File: Data Structures & Algorithms/string-encode-and-decode/submission-1.java
class Solution {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder encodedStr = new StringBuilder();
        for (String s : strs) {
            encodedStr.append(s.length()).append("#").append(s);
        }
        return encodedStr.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String str) {
        List<String> result = new ArrayList<>();
        int i = 0;
        while (i < str.length()) {
            int delimiterIndex = str.indexOf('#', i);
            int length = Integer.parseInt(str.substring(i, delimiterIndex));
            i = delimiterIndex + 1;
            result.add(str.substring(i, i + length));
            i += length;
        }
        return result;
    }
}